### PR TITLE
Enable CONCURRENT_TOKEN_V2 flag

### PR DESCRIPTION
**Pull request by igor-aptos**
### Description

Enable concurrent Token V2 structs from AIP-43.

We can do this now, that indexer change has landed. (which was blocking it)

(they will all be sequential until aggregator flag itself is enabled,
but this allows for new codepath to be executed and tested, and replay-verify to test aggregators)

### CODE REVIEW

**Comment by igor-aptos** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11932?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #11926** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11926?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #11932** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11932?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #11912** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11912?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11932?utm_source=stack-comment).

**Review by msmouse — APPROVED** [review]

File: execution/executor-benchmark/src/lib.rs [hunk 1]
```diff
@@ -746,7 +759,7 @@ mod tests {
         AptosVM::set_processed_transactions_detailed_counters();
         NativeExecutor::set_concurrency_level_once(4);
         test_generic_benchmark::<AptosVM>(
-            Some(TransactionTypeArg::ModifyGlobalResourceAggV2),
+            Some(TransactionTypeArg::ResourceGroupsGlobalWriteTag1KB),
```
Comment (georgemitenkov) [inline execution/executor-benchmark/src/lib.rs:762]: Was it renamed? branch not based on master so cannot tell

File: aptos-move/vm-genesis/src/lib.rs [hunk 2]
```diff
@@ -438,6 +438,7 @@ pub fn default_features() -> Vec<FeatureFlag> {
         FeatureFlag::SINGLE_SENDER_AUTHENTICATOR,
         FeatureFlag::SPONSORED_AUTOMATIC_ACCOUNT_V1_CREATION,
         FeatureFlag::FEE_PAYER_ACCOUNT_OPTIONAL,
+        FeatureFlag::CONCURRENT_TOKEN_V2,
```
Comment (georgemitenkov) [inline aptos-move/vm-genesis/src/lib.rs:441]: The ordering is intended?

**Review by georgemitenkov — APPROVED** [review]

**Comment by vusirikala** [comment]
Seems like this PR contains the code from https://github.com/aptos-labs/aptos-core/pull/11926 as well. Maybe a cleanup is required?

**Comment by vusirikala** [comment]
Do we also need to add the flag in release.yaml?

**Comment by igor-aptos** [comment]
when branch is cut is when release.yaml is configured.

updated the PR to not show stacked changes

File: aptos-move/vm-genesis/src/lib.rs (hunk 2 above)
Comment (igor-aptos) [reply aptos-move/vm-genesis/src/lib.rs:441]: order is just how features are defined in the enum. any order works the same

**Pull request merged**